src/ui/components/utils/notebook-snapshot.js

```diff
@@ -0,0 +1,135 @@
+        hint: "Please select a Notebook",
+        sections: [{
+            rows: [
+                {
+                    name: 'Entry',
+                    key: 'entry',
+                    control: 'textarea',
+                    required: false,
+                    cssClass: "l-textarea-sm"
+                },
+                {
+                    name: 'Snap Preview',
+                    key:"snapPreview",
+                    control: "snap-view",
+                    cssClass: "l-textarea-sm",
+                    src: url
+                },
+                {
+                    name: 'Save in Notebook',
+                    key: 'notebook',
+                    control: 'locator',
+                    validate: validateLocation
+                }
+            ]
+        }]
+    };
+
+    var overlayModel = {
+        title: taskForm.name,
+        message: 'Notebook Snapshot',
+        structure: taskForm,
+        value: {'entry': ""}
+    };
+
+    function validateLocation(newParentObj) {
+        return newParentObj.model.type === 'notebook';
+    }
+
+    return overlayModel;
+}
+
+function saveSnapShot(dialogService, imageUrl, imageType, imageSize, embedObject) {
+    let taskForm = generateTaskForm(imageUrl);
+
+    dialogService.getDialogResponse(
+        'overlay-dialog',
+        taskForm,
+        () => taskForm.value
+    ).then(options => {
+        let snapshotObject = {
+            src: options.snapPreview || imageUrl,
+            type: imageType,
+            size: imageSize
+        };
+
+        options.notebook.useCapability('mutation', function (model) {
+            var date = Date.now();
+
+            model.entries.push({
+                id: 'entry-' + date,
+                createdOn: date,
+                text: options.entry,
+                embeds: [{
+                    name: embedObject.name,
+                    cssClass: embedObject.cssClass,
+                    type: embedObject.id,
+                    id: 'embed-' + date,
+                    createdOn: date,
+                    snapshot: snapshotObject
+                }]
+            });
+        });
+    });
+}
+
+class NotebookSnapshot {
+    constructor(openmct) {
+        this.openmct = openmct;
+
+        this.capture = this.capture.bind(this);
+    }
+
+    capture(domainObject, DomElement) {
+        let exportImageService = this.openmct.$injector.get('exportImageService'),
+            dialogService = this.openmct.$injector.get('dialogService');
+
+        let type = this.openmct.types.get(domainObject.type),
+            embedObject = {
+                id: domainObject.identifier.key,
+                cssClass: type.cssClass,
+                name: domainObject.name
+            };
+
+        DomElement.classList.add('s-status-taking-snapshot');
+
+        exportImageService.exportPNGtoSRC(DomElement).then(function (blob) {
+            DomElement.classList.remove('s-status-taking-snapshot');
+
+            if (blob) {
```

Comment:
Is there a scenario where we don't get a blob back? If so, we should reject the promise.